python/rapidsmpf/rapidsmpf/examples/dask.py

```
@@ -184,6 +196,20 @@ def dask_cudf_shuffle(
         {"on": on, "column_names": list(df0.columns)},
         *sort_boundary_names,
     )
+    if cluster_kind == "single":
+        graph = single_rapidsmpf_shuffle_graph(*shuffle_graph_args)
+    else:
+        try:
+            from rapidsmpf.integrations.dask.shuffler import rapidsmpf_shuffle_graph
+
+            graph = rapidsmpf_shuffle_graph(*shuffle_graph_args)
+        except (ImportError, ValueError):
```

I've gone back and forth, but I *think* that I dislike the "auto" behavior here :)

I'm just not sure how many cases we have where we'd be fine with silently falling back to single-worker execution if you happen to mess up the install of rapidsmpf, or forget to start a Dask cluster, or if `rapidsmpf_shuffle_graph` happens to throw a `ValueError` for some new reason (*hopefully* it's just because you forgot to start a cluster, but with exceptions it's hard to know for sure). They just seem so drastically different that I'd want to know if something happened. I guess that argues for me specifying `cluster_kind="distributed"` but since it's an optional keyword I know I (and probably users) will forget to set it.

The upside of having a single entrypoint for shuffling dask stuff is appealing. But right now I'm just not sure it's worth the potential downside...

---

It could be worth considering that I don't actually want anyone **using** this function is practice (at least not yet).

With that said: Since it's a "DataFrame" API, it definitely feels "off" for the user to need to specify whether or not we are executing in "distributed mode". I don't think the `cluster_kind` argument really **need** to exist. If there is a distributed client available, we should run in distributed mode. Otherwise, we use "single".

The only reason the `cluster_kind` argument makes sense, is because some users may want to use "single" execution with `.compute(scheduler="synchronous")` when a distributed client happens to be active. Perhaps the argument could be `force_single: bool = False`?

---

Ah, I'd missed that this is in `rapidsmpf.examples`. In that case, maybe this doesn't matter either way.

---

I added a note to the doc-string that this API is currently for demonstration and testing purposes, so I think we are good on this front for now.